search/search_quality/sample.cpp

```diff
@@ -133,35 +133,37 @@ bool Sample::operator==(Sample const & rhs) const
 }
 
 // static
-bool Sample::DeserializeFromJSON(string const & jsonStr, std::vector<Sample> & samples)
+bool Sample::DeserializeFromJSONLines(string const & lines, std::vector<Sample> & samples)
 {
-  try
-  {
-    my::Json root(jsonStr.c_str());
-    if (!json_is_array(root.get()))
-      MYTHROW(my::Json::Exception, ("The field", "samples", "must contain a json array."));
-    size_t numSamples = json_array_size(root.get());
-    samples.resize(numSamples);
-    for (size_t i = 0; i < numSamples; ++i)
-      samples[i].DeserializeFromJSONImpl(json_array_get(root.get(), i));
-    return true;
-  }
-  catch (my::Json::Exception const & e)
+  istringstream is(lines);
+  string line;
+  vector<Sample> result;
+
+  while (getline(is, line))
   {
-    LOG(LERROR, ("Can't parse samples:", e.Msg(), jsonStr));
+    if (line.empty())
+      continue;
+
+    Sample sample;
+    if (!sample.DeserializeFromJSON(line))
+      return false;
+    result.emplace_back(move(sample));
   }
-  return false;
+
+  samples.swap(result);
+  return true;
 }
 
 // static
-void Sample::SerializeToJSON(std::vector<Sample> const & samples, std::string & jsonStr)
+void Sample::SerializeToJSONLines(std::vector<Sample> const & samples, std::string & lines)
 {
-  auto array = my::NewJSONArray();
   for (auto const & sample : samples)
-    json_array_append_new(array.get(), sample.SerializeToJSON().release());
-  std::unique_ptr<char, FreeDeletor> buffer(
-      json_dumps(array.get(), JSON_COMPACT | JSON_ENSURE_ASCII));
-  jsonStr.assign(buffer.get());
+  {
```

Comment:
then there is no symmetry with DeserializeFromJSONLines because DeserializeFromJSONLines. same points about pre-filled data are not in DeserializeFromJSONLines.